internal/bminventory/inventory.go

```diff
@@ -2057,6 +2069,56 @@ func optionalParam(data *string, field string, updates map[string]interface{}) {
 	}
 }
 
+func (b *bareMetalInventory) reportUsage(enabled bool, name string, props *map[string]interface{}, usages map[string]models.Usage) {
+	if enabled {
+		b.usageApi.AddUsage(usages, name, props)
+	} else {
+		b.usageApi.RemoveUsage(usages, name)
+	}
+}
+
+func (b *bareMetalInventory) reportProxyUsage(params *models.ClusterUpdateParams, usages map[string]models.Usage) {
+	props := map[string]interface{}{}
+	enabled := false
+	if params.HTTPProxy != nil {
+		props["http_proxy"] = 1
+		enabled = true
+	}
+	if params.HTTPSProxy != nil {
+		props["https_proxy"] = 1
+		enabled = true
+	}
+	if params.NoProxy != nil {
+		props["no_proxy"] = true
+		enabled = true
+	}
+	if enabled {
+		b.usageApi.AddUsage(usages, usage.ProxyUsage, &props)
+	} else {
+		b.usageApi.RemoveUsage(usages, usage.ProxyUsage)
+	}
+}
+
+func (b *bareMetalInventory) reportOperatorsUsage(updateOLMOperators []*models.MonitoredOperator, usages map[string]models.Usage) {
+	for _, operator := range updateOLMOperators {
+		b.usageApi.AddUsage(usages, operator.Name, nil)
+	}
+
+	allOlmOperators := b.operatorManagerApi.GetSupportedOperatorsByType(models.OperatorTypeOlm)
```

Comment:
Please add a comment describing these next lines - it's not obvious

Reply:
code changed to be self explanatory